File: my-folder/0238-product-of-array-except-self/solution.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        zero = nums.count(0)
        n = len(nums)
        mul = 1
        result = [0] * n
        mul = 1
        
        if zero > 1:
            return result
        if zero == 0:
            for num in nums:
                mul *= num
            for i in range(n):
                result[i] = mul // nums[i]  
        if zero == 1:
            for num in nums:
                if num != 0:
                    mul *= num

            for i in range(n):
                if nums[i] == 0:
                    result[i] = mul 
                else:
                    result[i] = 0 

        return result

File: my-folder/0238-product-of-array-except-self/solution.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        int zeros = 0;
        int n = nums.size();
        vector<int> res(n, 0);

        for (int i = 0; i < n; i++) {
            if (nums[i] == 0) {
                zeros += 1;
            }
        }

        if (zeros > 1 || n <= 1) {
            return res;
        }

        if (zeros == 0) {
            int mul = 1;
            for (int j = 0; j < n; j++) {
                mul *= nums[j];
            }
            for (int k = 0; k < n; k++) {
                res[k] = mul / nums[k];
            }
        }

        if (zeros == 1) {
            int mul = 1;
            int i = 0;
            for(int i=0; i<n; i++) {
                if(nums[i] != 0) mul *= nums[i];
            }
            for (int j = 0; j < n; j++) {
                if (nums[j] == 0) {
                    res[j] = mul;
                } else {
                    res[j] = 0;
                }
            }
        }

        return res;
    }
};

